lib/cache/fetchAndExtract.js

```diff
@@ -18,6 +15,8 @@ module.exports = async ({ url, name, tmpLoc, outLoc, raw, requestOpts }) => {
             })
             .on('warn', reporter.warn)
 
+        await fs.ensureDir(outLoc)
```

Comment:
I think this works here, and works great for the **archive** download case!

However there may be an issue if the `outLoc` doesn't have an extension - since we don't use `fs.move` in the second case (non-archives) I believe the writeStream may fail if this `ensureDir` naively creates the actual `outLoc` (not `path.dirname(outLoc)`) before the download starts.  Does that check out?  It *might* be OK since the "raw" case should always be a file, not a directory, but still seems like it could lead to confusion?

Reply:
What's a non-archive in this case?

Reply:
https://github.com/dhis2/cli-helpers-engine/pull/16/files#diff-2b5697f99fbad1dde1f214141572ca82R55

The `else` case, which downloads a file directly to `outLoc` rather than downloading an archive (`.tar[.gz]`) to `tmpLoc` and copying it with `fs.move` and `{ overwrite: true }` to `outLoc`

Reply:
Fixed now.